bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/GetListOfEntriesOfALedgerProcessorV3.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.bookkeeper.proto.BookkeeperProtocol.GetListOfEntriesOfALedgerRequest;
+import org.apache.bookkeeper.proto.BookkeeperProtocol.GetListOfEntriesOfALedgerResponse;
+import org.apache.bookkeeper.proto.BookkeeperProtocol.Request;
+import org.apache.bookkeeper.proto.BookkeeperProtocol.Response;
+import org.apache.bookkeeper.proto.BookkeeperProtocol.StatusCode;
+import org.apache.bookkeeper.util.MathUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A processor class for v3 entries of a ledger packets.
+ */
+public class GetListOfEntriesOfALedgerProcessorV3 extends PacketProcessorBaseV3 implements Runnable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GetListOfEntriesOfALedgerProcessorV3.class);
+    protected final GetListOfEntriesOfALedgerRequest getListOfEntriesOfALedgerRequest;
+    protected final long ledgerId;
+
+    public GetListOfEntriesOfALedgerProcessorV3(Request request, Channel channel,
+            BookieRequestProcessor requestProcessor) {
+        super(request, channel, requestProcessor);
+        this.getListOfEntriesOfALedgerRequest = request.getGetListOfEntriesOfALedgerRequest();
+        this.ledgerId = getListOfEntriesOfALedgerRequest.getLedgerId();
+    }
+
+    private GetListOfEntriesOfALedgerResponse getListOfEntriesOfALedgerResponse() {
+        long startTimeNanos = MathUtils.nowInNano();
+
+        GetListOfEntriesOfALedgerResponse.Builder getListOfEntriesOfALedgerResponse = GetListOfEntriesOfALedgerResponse
+                .newBuilder();
+        getListOfEntriesOfALedgerResponse.setLedgerId(ledgerId);
+
+        if (!isVersionCompatible()) {
+            getListOfEntriesOfALedgerResponse.setStatus(StatusCode.EBADVERSION);
+            requestProcessor.getRequestStats().getGetListOfEntriesOfALedgerStats()
+                    .registerFailedEvent(MathUtils.elapsedNanos(startTimeNanos), TimeUnit.NANOSECONDS);
+            return getListOfEntriesOfALedgerResponse.build();
+        }
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Received new getListOfEntriesOfALedger request: {}", request);
+        }
+        StatusCode status = StatusCode.EOK;
+        ByteBuf listOfEntriesOfALedger = null;
+        try {
+            listOfEntriesOfALedger = requestProcessor.bookie.getListOfEntriesOfALedger(ledgerId);
+            if (listOfEntriesOfALedger != null) {
+                getListOfEntriesOfALedgerResponse
+                        .setAvailabilityOfEntriesOfLedger(ByteString.copyFrom(listOfEntriesOfALedger.nioBuffer()));
+            }
+        } catch (Bookie.NoLedgerException e) {
+            status = StatusCode.ENOLEDGER;
+            LOG.error("No ledger found while performing getListOfEntriesOfALedger from ledger: {}", ledgerId, e);
+        } catch (IOException e) {
+            status = StatusCode.EIO;
+            LOG.error("IOException while performing getListOfEntriesOfALedger from ledger: {}", ledgerId);
```

Comment:
any reason for not reporting the stacktrace of the exception
IOException are very generic and having the detail will ease debug

Reply:
will fix it